server.go

```diff
@@ -176,20 +176,23 @@ func (s Server) parseRequestParams(r *http.Request, refSchema schema.Schema, ref
 		return ListRequestParams{}, &scimErr
 	}
 
-	rawFilter := strings.TrimSpace(r.URL.Query().Get("filter"))
-	decodedFilter, _ := url.QueryUnescape(rawFilter)
 	var validator f.Validator
```

Comment:
Why do we need this declaration?

Reply:
> You mean the validator? It contains the filter expression.

I see, I was a bit confused by the conditional mutation here. Maybe you can extract some kind of function that returns a validator for a specific request?